drivers/wdm/audio/backpln/portcls/undoc.cpp

```diff
@@ -281,14 +273,14 @@ PropertyItemDispatch(
     if (PropertyRequest->PropertyItem && PropertyRequest->PropertyItem->Handler)
     {
         // now call the handler
-        UNICODE_STRING GuidBuffer;
-        RtlStringFromGUID(Property->Set, &GuidBuffer);
-        DPRINT("Calling Node %lu MajorTarget %p MinorTarget %p PropertySet %S PropertyId %lu PropertyFlags %lx InstanceSize %lu ValueSize %lu Handler %p PropertyRequest %p PropertyItemFlags %lx PropertyItemId %lu\n",
-                PropertyRequest->Node, PropertyRequest->MajorTarget, PropertyRequest->MinorTarget, GuidBuffer.Buffer, Property->Id, Property->Flags, PropertyRequest->InstanceSize, PropertyRequest->ValueSize,
-                PropertyRequest->PropertyItem->Handler, PropertyRequest, PropertyRequest->PropertyItem->Flags, PropertyRequest->PropertyItem->Id);
-        RtlFreeUnicodeString(&GuidBuffer);
```

Comment:
What line?

Reply:
RtlFreeUnicodeString(&GuidBuffer);

Reply:
I can't find it.

Reply:
It's one of the lines that was removed completely. Here it is in the diff:

```patch
--- a/drivers/wdm/audio/backpln/portcls/undoc.cpp
+++ b/drivers/wdm/audio/backpln/portcls/undoc.cpp
@@ -281,14 +273,14 @@ PropertyItemDispatch(
     if (PropertyRequest->PropertyItem && PropertyRequest->PropertyItem->Handler)
     {
         // now call the handler
-        UNICODE_STRING GuidBuffer;
-        RtlStringFromGUID(Property->Set, &GuidBuffer);
-        DPRINT("Calling Node %lu MajorTarget %p MinorTarget %p PropertySet %S PropertyId %lu PropertyFlags %lx InstanceSize %lu ValueSize %lu Handler %p PropertyRequest %p PropertyItemFlags %lx PropertyItemId %lu\n",
-                PropertyRequest->Node, PropertyRequest->MajorTarget, PropertyRequest->MinorTarget, GuidBuffer.Buffer, Property->Id, Property->Flags, PropertyRequest->InstanceSize, PropertyRequest->ValueSize,
-                PropertyRequest->PropertyItem->Handler, PropertyRequest, PropertyRequest->PropertyItem->Flags, PropertyRequest->PropertyItem->Id);
-        RtlFreeUnicodeString(&GuidBuffer);                              <==============
+        //UNICODE_STRING GuidBuffer;
+        //RtlStringFromGUID(Property->Set, &GuidBuffer);
+        //DPRINT("Calling Node %lu MajorTarget %p MinorTarget %p PropertySet %S PropertyId %lu PropertyFlags %lx InstanceSize %lu ValueSize %lu Handler %p PropertyRequest %p PropertyItemFlags %lx PropertyItemId %lu\n",
+        //        PropertyRequest->Node, PropertyRequest->MajorTarget, PropertyRequest->MinorTarget, GuidBuffer.Buffer, Property->Id, Property->Flags, PropertyRequest->InstanceSize, PropertyRequest->ValueSize,
+        //        PropertyRequest->PropertyItem->Handler, PropertyRequest, PropertyRequest->PropertyItem->Flags, PropertyRequest->PropertyItem->Id);
+
         Status = PropertyRequest->PropertyItem->Handler(PropertyRequest);
-        DPRINT("Status %lx ValueSize %lu Information %lu\n", Status, PropertyRequest->ValueSize, Irp->IoStatus.Information);
+        //DPRINT("Status %lx ValueSize %lu Information %lu\n", Status, PropertyRequest->ValueSize, Irp->IoStatus.Information);
         Irp->IoStatus.Information = PropertyRequest->ValueSize;
 
         if (Status != STATUS_PENDING)
```

IMO it's better to delete this block entirely if it's not needed, but if we're keeping it we should also include the Free call, otherwise it's a memory leak waiting to happen.

Reply:
Isn't it commented out though?

Reply:
It looks deleted on my end, e.g. there is no corresponding

```patch
+        //RtlFreeUnicodeString(&GuidBuffer);
```

in the diff. Not sure why, perhaps we're looking at different versions of the file?

Reply:
JoseSanchezz is right. Either you should not touch anything in the diff-excerpt that he linked https://github.com/reactos/reactos/pull/3054#discussion_r471050442, or you should add the line that he mentioned in a commented way. I would prefer if you did not touch this entirely honestly (at least after having verified that your patch does still work when all that code is active).
Contrary if you would comment all those changes but strip that single line (like you did), then you make the commented code incomplete and wrong!

Reply:
I didn't modify the code that this review is about.

Reply:
Sure you did. Your first commit does. And the final state after commit1-5 still does.

Reply:
So, should I use the previous code or completely remove it?

Reply:
I vote for completely using the previous code here with the aim to minimize the unneeded diff, ideally after verifying that the code is still capable to fix those issues and does still compile (although I strongly believe that this additional heap allocation and debug print will not have any negative impact).

Reply:
Reverting that change didn't have any negative impacts. So, I fixed it.